test/unit/idx/src/MapIdxPitch.cpp

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2020 Jeffrey Kelling
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/idx/Accessors.hpp>
+#include <alpaka/idx/MapIdx.hpp>
+#include <alpaka/dev/Traits.hpp>
+#include <alpaka/mem/view/ViewPlainPtr.hpp>
+#include <alpaka/mem/view/ViewSubView.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#include <alpaka/test/dim/TestDims.hpp>
+#include <alpaka/test/Extent.hpp>
+
+#include <catch2/catch.hpp>
+
+//-----------------------------------------------------------------------------
+TEMPLATE_LIST_TEST_CASE( "mapIdxPitch", "[idx]", alpaka::test::dim::TestDims)
+{
+    using Dim = TestType;
+    using Idx = std::size_t;
+    using Vec = alpaka::vec::Vec<Dim, Idx>;
+
+    auto const extentNd(alpaka::vec::createVecFromIndexedFn<Dim, alpaka::test::CreateVecWithIdx<Idx>::template ForExtentBuf>());
+
+    using Acc = alpaka::example::ExampleDefaultAcc<Dim, Idx>;
+    using Dev = alpaka::dev::Dev<Acc>;
+    using Elem = int;
+    auto const devAcc = alpaka::pltf::getDevByIdx<Acc>(0u);
+    alpaka::mem::view::ViewPlainPtr<Dev, Elem, Dim, Idx> parentView( nullptr, devAcc, extentNd );
+
+    auto const offset(Vec::all(4u));
+    auto const extent(Vec::all(4u));
+    auto const idxNd(Vec::all(2u));
+    alpaka::mem::view::ViewSubView<Dev, Elem, Dim, Idx> view( parentView, extent, offset );
+    auto pitch = alpaka::mem::view::getPitchBytesVec(view);
+    for(Idx a = 0; a < Dim::value; ++a)
+        pitch[a] /= sizeof(Elem);
```

Comment:
I am not sure if we can make the assumption that the pitch is cleanly dividable by the element size.
The `Elem` type is something that is only known on the alpaka side. The allocation size is defined by `extent*sizeof(Elem)`. CUDA then may introduce some padding to make the pitch better match the hardware.
If we have a custom struct that has a size of 3 bytes and allocate a 2-dimenstional buffer of size 2x2, then CUDA may give us a buffer with pitch 8 instead of 6. However, 8 is not cleanly dividable by 3.

Reply:
The pitch expected by `MapIdxPitch` is not in bytes but in elements, the same unit the index is in, because it does not know the element size. Not sure if this problem should be solved inside `MapIdxPitch` or outside...

Reply:
MapIdxPitch does fundamentally not work with padded buffers: One could give it information about `TElem` so it could take the padding into account, but then it would have to return either an index in bytes, which would produce weird code in the caller, or return a pointer instead of an index, which would basically turn it into an iterator.